drivers/i2c/i2c_dw.c

```diff
@@ -275,8 +275,8 @@ static void i2c_dw_isr(const struct device *port)
 		}
 
 	} else {
-#ifdef CONFIG_I2C_SLAVE
-		const struct i2c_slave_callbacks *slave_cb = dw->slave_cfg->callbacks;
```

Comment:
General comment: Should we rename the `struct i2c_target_callback` instances as well all over?

Reply:
Yes absolutely, but this can be done over time by myself or driver authors, the goal in this PR is very explicit in correcting the public facing terminology. The drivers will require additionally combing over and cleaning up. A task can be created to cover that and it can be delegated to various driver authors hopefully.